app/web_conn.go

```diff
@@ -30,9 +31,9 @@ const (
 type WebConn struct {
 	WebSocket                 *websocket.Conn
 	Send                      chan model.WebSocketMessage
-	SessionToken              string
-	SessionExpiresAt          int64
-	Session                   *model.Session
+	sessionToken              atomic.Value
+	sessionExpiresAt          int64
```

Comment:
To use a 64 bit integer atomically, it needs to be 64-bit aligned:

> On both ARM and x86-32, it is the caller's responsibility to arrange for 64-bit alignment of 64-bit words accessed atomically. The first word in a variable or in an allocated struct, array, or slice can be relied upon to be 64-bit aligned.
– https://golang.org/pkg/sync/atomic/#pkg-note-BUG

You should move `sessionExpiresAt` to the top of the struct.

Reply:
Moved